### ISSUE TITLE
Strange issue running example

### ISSUE THREAD

**Issue by mauro3** [JuliaComputing/ArrayFire.jl#37]
When I run the example I get the following error:
```
~ >> julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.5 (2016-03-18 00:58 UTC)
 _/ |\__'_|_|_|\__'_|  |  
|__/                   |  x86_64-unknown-linux-gnu

julia> include(".julia/v0.4/ArrayFire/examples/blackscholes.jl")
Device[0] has no support for OpenGL Interoperation
ERROR: LoadError: MethodError: `call` has no method matching call(::Void, ::Type{ArrayFire.AFArray{Float32,N}}, ::Ptr{Void})
Closest candidates are:
  BoundsError(::Any...)
  TypeVar(::Any...)
  TypeConstructor(::Any...)
  ...
 in blackscholes_serial at /home/mauro/.julia/v0.4/ArrayFire/examples/blackscholes.jl:10
 in driver at /home/mauro/.julia/v0.4/ArrayFire/examples/blackscholes.jl:57
 in include at ./boot.jl:261
 in include_from_node1 at ./loading.jl:320
while loading /home/mauro/.julia/v0.4/ArrayFire/examples/blackscholes.jl, in expression starting on line 67
```

it goes away if I change the variable `call` to `call_` on lines https://github.com/JuliaComputing/ArrayFire.jl/blob/e90e9eb6b68c2a1948c70f64297e4f435fe9698e/examples/blackscholes.jl#L19-20

Maybe this is a Julia bug, so let me know if I should repost there.  Also, does anyone else see this?

**Comment by ranjanan** [comment]
Are you running this on Julia v0.4? Can you check to see if the same issue is there on v0.5?

**Comment by ranjanan** [comment]
The following code works:
```julia
a = rand(Float32, 10, 10)
b = AFArray(a)
```
but the following code errors out: 
```julia
a = rand(Float32, 10, 10)
call = 1
b = AFArray(a)
```
with `ERROR: invalid function object`

I do define a `call(::Type{AFArray})` [here](https://github.com/JuliaComputing/ArrayFire.jl/blob/master/src/create.jl#L31)

**Comment by mauro3** [comment]
On 0.5 it works.

**Comment by ranjanan** [comment]
Fine, I just changed `call` to `call_`. Fixed by 0be93d0374d02b27f26a325efae2a7d8f4a8ed04

**Comment by ranjanan** [comment]
@mauro3 I think this is why `call` is deprecated on v0.5 😄
